modules/substrate/src/lib.rs

```diff
@@ -117,37 +117,13 @@ decl_storage! {
 				.initial_header
 				.clone()
 				.expect("An initial header is needed");
-			let initial_hash = initial_header.hash();
 
-			<BestHeight<T>>::put(initial_header.number());
-			<BestHeaders<T>>::put(vec![initial_hash]);
-			<BestFinalized<T>>::put(initial_hash);
-
-			let authority_set =
-				AuthoritySet::new(config.initial_authority_list.clone(), config.initial_set_id);
-			CurrentAuthoritySet::put(authority_set);
-
-			let mut signal_hash = None;
-			if let Some(ref change) = config.first_scheduled_change {
-				assert!(
-					change.height > *initial_header.number(),
-					"Changes must be scheduled past initial header."
-				);
-
-				signal_hash = Some(initial_hash);
-				<NextScheduledChange<T>>::insert(initial_hash, change);
-			};
-
-			<ImportedHeaders<T>>::insert(
-				initial_hash,
-				ImportedHeader {
-					header: initial_header,
-					requires_justification: false,
-					is_finalized: true,
-					signal_hash,
-				},
+			initialize_bridge::<T>(
+				initial_header,
```

Comment:
Imo the `initial_header` and `initial_authority_list` must be `Option`s, or probably they should be removed from `GenesisConfig` (along with `initial_set_id` and `first_scheduled_change`). Because passing something that we call `millau_genesis_header` in the code, but we know is not genesis header of Millau is not good :)

Reply:
Probably let's add struct `InitializationData` and make it `Option`?